_posts/2019-12-06-Rust-für-Java-Entwickler-Teil-2.md

```diff
@@ -0,0 +1,308 @@
+```rust
+assert_eq!(resp.status(), http::StatusCode::NOT_FOUND);
+```
+testen.
+
+Nun wollen wir auch den Fall prüfen, dass eine Person mit der ID exisitert.
+Dann werden wir im Test erkennen, ob die ID, der Name und das Alter korrekt ist (also wie in der Testdatei).
+Wir erstellen dazu wieder unsere Test-App und den Request mit einer ID, die in unserer Testdatei existiert.
+
+```rust
+#[test]
+fn test_returns_success() {
+    let mut app = test::init_service(App::new().service(request_handler::get));
+    let req = test::TestRequest::get().uri("/persons/2").to_request();
+}
+```
+
+Im test-Modul von Actix-Web gibt es eine Funktion ```read_response_json```, die uns das deserialisierte Objekt unseres Requests liefert.
+Diese Funktion rufen wir wie folgt auf:
+
+```rust
+let result: person::Person = test::read_response_json(&mut app, req);
+```
+
+In der ```result```-Variable ist jetzt die Person Bob, die in der Datei die ID 2 hat.
+Zum Schluss prüfen wir jetzt noch die einzelnen Attribute und erhalten unseren kompletten Test:
+
+```rust
+#[test]
+fn test_returns_success() {
+    let mut app = test::init_service(App::new().service(request_handler::get));
+    let req = test::TestRequest::get().uri("/persons/2").to_request();
+
+    let result: person::Person = test::read_response_json(&mut app, req);
+
+    assert_eq!(result.id(), 2);
+    assert_eq!(result.age(), 24);
+    assert_eq!(result.name(), "Bob");
+}
+```
+
+Diejenigen, die sich noch weiter mit den Möglichkeiten von Actix-Web zur Fehlberhandlung auseinandersetzen wollen, können sich [hier](https://actix.rs/docs/testing/) weiter in das Thema einlesen.
+
+
+# Fazit
+
+In diesem Teil der Artikel-Serie haben wir gesehen, welche Möglichkeiten Actix-Web für die Fehlerbehandlung und das automatische Testen bietet.
+Auch hier habe ich natürlich nicht die komplette Funktionspalette des Frameworks vorgestellt.
+Doch wir haben einen guten Einblick in die generellen Konzepte  und Funktionsweisen bekommen.
+Wer tiefer in das Framework einsteigen will, kann sich an der [offiziellen Dokumentation](https://actix.rs/docs/) orientieren.
+Dort wird auch auf die komplette API-Dokumentation verwiesen.
+
+Ein Beispiel für eine vollständige App mit Actix ist z.B. [hier](https://github.com/fairingrey/actix-realworld-example-app) zu finden.
+
+Der in diesem Artikel entwickelte Code ist [auf Github](https://github.com/KilianKrause/rest-api-with-actix) verfügbar.
+Der Code ist an einigen Stellen optimiert bzw. modifiziert worden.
+
+Ist Actix-Web ein Kandidat für die Entwicklung eines echten Webservices?
+Meiner Meinung nach ist die Erstellung der REST-API mit Actix-Web sehr intuitiv und mir gefällt das Framework sehr gut.
+Allerdings spielen bei der Entscheidung auch andere Faktoren eine Rolle, wie z.B. ORM-Software, Kommunikation mit anderen Systemen u.v.m.
+Da gibt es in Rust noch viel zu tun und die Java/Spring-Welt ist einfach ausgereifter - was auch verständlich ist, denn Rust ist noch sehr jung.
```

Comment:
Finde den Satz nicht so gut.
Java & Spring sind in diesen Bereichen ausgereifter und Rust muss hier noch viel tun - was auch..